concordium-consensus/src/Concordium/GlobalState/Persistent/Accounts.hs

```diff
@@ -168,11 +169,11 @@ getAccount addr Accounts{..} = AccountMap.lookup addr accountMap >>= \case
 -- |Retrieve an account associated with the given credential registration ID.
 -- Returns @Nothing@ if no such account exists.
 getAccountByCredId :: (MonadBlobStore m, IsProtocolVersion pv) => ID.CredentialRegistrationID -> Accounts pv -> m (Maybe (AccountIndex, PersistentAccount (AccountVersionFor pv)))
-getAccountByCredId cid accs@Accounts{accountRegIds = Null,..} = Trie.lookup cid accountRegIdHistory  >>= \case
+getAccountByCredId cid accs@Accounts{accountRegIds = Null,..} = Trie.lookup (encode cid) accountRegIdHistory  >>= \case
```

Comment:
The type of this function could be changed to take a raw credential registration ID with no loss.

The function is only used in queries.

Reply:
After looking at how `getAccountByCredId` is used, I'm not sure that this change will spare any unneeded conversions.

Reply:
I believe `getAccountByCredId` is only used for external queries, i.e., not anything consensus related. There, we can just parse the input as a raw credential ID. If it is invalid, the query will in any case fail to find any account. So there is no loss I believe.

Reply:
That's exactly my concern. If we change the type of `getAccountByCredId` to avoid a single credential ID conversion, then the structure of `AccountIdentifier` should change too. This will change the external consensus API.

Even if the change will only lead to accounts not being found if an invalid ID is passed to the query, I have a hunch someone somewhere will get upset about it, if the change isn't properly communicated.

Reply:
I am not exactly sure which scenario you mean.

The behaviour of the query will be exactly the same. It will return the same responses, and it will fail in the same scenarios (i.e., return Null if it does not find it).